### PR TITLE
feat: create Target Group and ALB for documentation

### LINKED ISSUE #123: Add target group for documentation
We will soon deploy https://github.com/cds-snc/notification-documentation to documentation.notification.canada.ca so we need:
- a target group for this
- a listener rule to forward to the new target group
- a redirection for doc.notification.canada.ca to documentation.notification.canada.ca (similar to the WWW redirection)

File to look at https://github.com/cds-snc/notification-terraform/blob/8a859b0028597f6a90a115edaefc6d85a54536a6/aws/eks/alb.tf#L174-L203

### PR DESCRIPTION
**Pull request by RenDavidDimen**
# Summary | Résumé

Adding the new target group for the documentation site we are putting up. I've also created some rules for the load balancer based off similar logic within the repo.

Fixes #123 
Trello card: https://github.com/cds-snc/notification-terraform/pull/139

### CODE REVIEW

File: aws/eks/alb.tf [hunk 1]
```diff
@@ -279,3 +279,45 @@ resource "aws_wafv2_web_acl_association" "notification-canada-ca" {
   web_acl_arn  = aws_wafv2_web_acl.notification-canada-ca.arn
 }
 
+###
+# Documentation Specific Routing
+###
```
Comment (AntoineAugusti) [inline aws/eks/alb.tf:284]: Could you move this before the WAF?

File: aws/eks/alb.tf [hunk 2]
```diff
@@ -279,3 +279,45 @@ resource "aws_wafv2_web_acl_association" "notification-canada-ca" {
...
+resource "aws_alb_target_group" "notification-canada-ca-documentation" {
+  name     = "notification-can-documentation"
```
Comment (AntoineAugusti) [inline aws/eks/alb.tf:286]: It seems like we prefix names with `notification-canada-ca` or `notification-` so pick one of these but avoid the `can` part

**Review by AntoineAugusti — CHANGES_REQUESTED** [review]
Good work! 👏  2 tiny things to change and then this will be good to go

File: aws/eks/alb.tf (hunk 1 above)
Comment (RenDavidDimen) [reply aws/eks/alb.tf:275]: Moved Documentation Section before WAF

File: aws/eks/alb.tf (hunk 2 above)
Comment (RenDavidDimen) [reply aws/eks/alb.tf]: Updated name to match prefix norms 👍

**Review by AntoineAugusti — APPROVED** [review]
Good work! 🚢

**Comment by AntoineAugusti** [comment]
Fixes #123

**Comment by AntoineAugusti** [comment]
@RenDavidDimen After merging this PR, could you give out to @jimleroyer the new target group's ARN? He needs it for https://github.com/cds-snc/notification-manifests/pull/111

**Pull request merged**